packages/strapi-plugin-content-type-builder/admin/src/containers/LeftMenu/index.js

```diff
@@ -86,26 +85,26 @@ function LeftMenu({ wait }) {
     );
   };
 
-  const handleClickOpenModal = async type => {
+  const handleClickOpenModal = async (type, kind = '') => {
     if (canOpenModalCreateCTorComponent()) {
-      const eventName =
-        type === 'contentType'
-          ? 'willCreateContentType'
-          : 'willCreateComponent';
-
-      emitEvent(eventName);
+      emitEvent(
+        `willCreate${upperFirst(
+          camelCase(kind === 'singleType' ? kind : type)
+        )}`
+      );
 
       await wait();
       push({
-        search: `modalType=${type}&actionType=create&settingType=base&forTarget=${type}&headerId=${getTrad(
+        search: `modalType=${type}${
```

Comment:
Here you can use the utility `makeSearch` so you can use an object which makes the code a bit cleaner.